core/src/repair_service.rs

```diff
@@ -953,6 +987,7 @@ mod test {
             shreds.pop();
 
             blockstore.insert_shreds(shreds, None, false).unwrap();
+            std::thread::sleep(Duration::from_millis(DEFAULT_MS_PER_SLOT) + DEFER_REPAIR_THRESHOLD);
```

Comment:
Can you help me understand all of the sleeps that were inserted into the tests?

Reply:
The sleeps are to ensure that the propagation delay threshold has been reached. An alternative would be to have the propagation check be optional via a parameter, but this would only be used for tests.

Reply:
Maybe we can add a function like:
```
// Sleeps until all shreds in this slot are eligible for repair
fn wait_for_repair_threshold(defer_repair_threshold: Duration) {
    std::thread::sleep(Duration::from_millis(DEFAULT_MS_PER_SLOT) + defer_repair_threshold);
}
```

Reply:
+1 to Brennan's suggestion.

but ideally we shouldn't be adding so many `sleep` in the code.

Reply:
timing values are now passed as parameters. `sleep`s removed.